### PR TITLE
Add flow types RNTester examples

### PR DESCRIPTION
**Pull request by rickhanlonii**
This PR adds flow types for the RNTester examples, and updates all of the RNTester examples to match the flow type consistently.

Previously, there was a mix of static class definitions and whether or not pages exported examples or a component. Now we will always export the same way, enforced by flow types

Note: I also fixed most of the $FlowFixMe in changed components

Test Plan:
----------
flow check

### CODE REVIEW

File: RNTester/js/PickerExample.js [hunk 1]
```diff
@@ -21,10 +21,6 @@ const {Picker, Text} = ReactNative;
 const Item = Picker.Item;
 
 class PickerExample extends React.Component<{}, $FlowFixMeState> {
```
Comment (rickhanlonii) [inline RNTester/js/PickerExample.js]: This change will conflict with https://github.com/facebook/react-native/pull/22828 but should be the only conflict

File: RNTester/js/CheckBoxExample.js [hunk 2]
```diff
@@ -48,7 +53,12 @@ class DisabledCheckBoxExample extends React.Component<{}, $FlowFixMeState> {
   }
 }
 
-class EventCheckBoxExample extends React.Component<{}, $FlowFixMeState> {
+type EventState = {|
+  eventCheckBoxIsOn: boolean,
+  eventCheckBoxRegressionIsOn: boolean,
+|};
+
+class EventCheckBoxExample extends React.Component<{}, EventState> {
```
Comment (RSNara) [inline RNTester/js/CheckBoxExample.js]:
```typescript
type EventProps = $ReadOnly<{||}>
class EventCheckBoxExample extends React.Component<EventProps, EventState> {
```

File: RNTester/js/ClipboardExample.js [hunk 3]
```diff
@@ -14,7 +14,11 @@ const React = require('react');
 const ReactNative = require('react-native');
 const {Clipboard, View, Text, StyleSheet} = ReactNative;
 
-class ClipboardExample extends React.Component<{}, $FlowFixMeState> {
+type State = {|
+  content: string,
+|};
+
+class ClipboardExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/ClipboardExample.js]:
```typescript
type Props = $ReadOnly<{||}>
class ClipboardExample extends React.Component<Props, State> {
```

File: RNTester/js/DatePickerAndroidExample.js [hunk 4]
```diff
@@ -21,10 +21,24 @@ const {
...
+  spinnerText: string,
+  calendarText: string,
+  defaultText: string,
+  minText: string,
+  maxText: string,
+  presetText: string,
+  allText: string,
+|};
 
+class DatePickerAndroidExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/DatePickerAndroidExample.js]:
```typescript
type Props = $ReadOnly<{||}>;
class DatePickerAndroidExample extends React.Component<Props, State> {
```

File: RNTester/js/FlatListExample.js [hunk 5]
```diff
@@ -40,10 +42,19 @@ const VIEWABILITY_CONFIG = {
...
+  horizontal: boolean,
+  inverted: boolean,
+  filterText: string,
+  fixedHeight: boolean,
+  logViewable: boolean,
+  virtualized: boolean,
+  empty: boolean,
+|};
 
+class FlatListExample extends React.PureComponent<{}, State> {
```
Comment (RSNara) [inline RNTester/js/FlatListExample.js]:
```typescript
type Props = $ReadOnly<{||}>
class FlatListExample extends React.PureComponent<Props, State> {
```

File: RNTester/js/InputAccessoryViewExample.js [hunk 6]
```diff
@@ -23,7 +23,7 @@ const {
   View,
 } = ReactNative;
 
-class Message extends React.PureComponent<*> {
+class Message extends React.PureComponent<{}> {
```
Comment (RSNara) [inline RNTester/js/InputAccessoryViewExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/KeyboardAvoidingViewExample.js [hunk 7]
```diff
@@ -25,11 +25,12 @@ const {
...
-class KeyboardAvoidingViewExample extends React.Component {
-  static title = '<KeyboardAvoidingView>';
-  static description =
-    'Base component for views that automatically adjust their height or position to move out of the way of the keyboard.';
+type State = {|
+  behavior: string,
+  modalOpen: boolean,
+|};
 
+class KeyboardAvoidingViewExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/KeyboardAvoidingViewExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/MaskedViewExample.js [hunk 8]
```diff
@@ -22,11 +22,11 @@ const {
...
 
-class MaskedViewExample extends React.Component<{}, $FlowFixMeState> {
-  static title = '<MaskedViewIOS>';
-  static description =
-    'Renders the child view with a mask specified in the `renderMask` prop.';
+type State = {|
+  alternateChildren: boolean,
+|};
 
+class MaskedViewExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/MaskedViewExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/ShareExample.js [hunk 9]
```diff
@@ -14,19 +14,11 @@ const React = require('react');
...
-      return <ShareMessageExample />;
-    },
-  },
-];
+type State = {|
+  result: string,
+|};
 
-class ShareMessageExample extends React.Component<$FlowFixMeProps, any> {
+class ShareMessageExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/ShareExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/TVEventHandlerExample.js [hunk 10]
```diff
@@ -15,28 +15,9 @@ const ReactNative = require('react-native');
...
-  {
-    lastEventType: string,
-  },
-> {
-  /* $FlowFixMe(>=0.85.0 site=react_native_fb) This comment suppresses an error
-   * found when Flow v0.85 was deployed. To see the error, delete this comment
-   * and run Flow. */
-  constructor(props) {
+type State = {|lastEventType: string|};
+class TVEventHandlerView extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/TVEventHandlerExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/TabBarIOSExample.js [hunk 11]
```diff
@@ -17,11 +17,13 @@ const {StyleSheet, TabBarIOS, Text, View} = ReactNative;
...
-  static title = '<TabBarIOS>';
-  static description = 'Tab-based navigation.';
-  static displayName = 'TabBarExample';
+type State = {|
+  selectedTab: string,
+  notifCount: number,
+  presses: number,
+|};
 
+class TabBarExample extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/TabBarIOSExample.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/ToastAndroidExample.android.js [hunk 12]
```diff
@@ -17,12 +17,7 @@ const {StyleSheet, Text, ToastAndroid, TouchableWithoutFeedback} = ReactNative;
 const RNTesterBlock = require('RNTesterBlock');
 const RNTesterPage = require('RNTesterPage');
 
-class ToastExample extends React.Component<{}, $FlowFixMeState> {
-  static title = 'Toast Example';
-  static description =
-    'Example that demonstrates the use of an Android Toast to provide feedback.';
-  state = {};
-
+class ToastExample extends React.Component<{}> {
```
Comment (RSNara) [inline RNTester/js/ToastAndroidExample.android.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/ViewPagerAndroidExample.android.js [hunk 13]
```diff
@@ -33,7 +33,8 @@ const IMAGE_URIS = [
   'https://apod.nasa.gov/apod/image/1510/lunareclipse_27Sep_beletskycrop4.jpg',
 ];
 
-class LikeCount extends React.Component {
+type State = {|likes: number|};
+class LikeCount extends React.Component<{}, State> {
```
Comment (RSNara) [inline RNTester/js/ViewPagerAndroidExample.android.js]: Same deal here. Should probably constrain the props.

File: RNTester/js/InputAccessoryViewExample.js [hunk 14]
```diff
@@ -108,4 +104,14 @@ const styles = StyleSheet.create({
...
 });
 
-module.exports = InputAccessoryViewExample;
+exports.title = '<InputAccessoryView>';
+exports.description =
+  'Example showing how to use an InputAccessoryView to build an iMessage-like sticky text input';
+exports.examples = [
+  {
+    title: 'Simple view with sticky input',
+    render: function(): React.Element<any> {
```
Comment (RSNara) [inline RNTester/js/InputAccessoryViewExample.js]: We could probably just use `React.Node` here.

Comment (rickhanlonii) [reply RNTester/js/InputAccessoryViewExample.js]: Fixed here, should this be consistent elsewhere? I can follow up with another PR is so

**Comment by rickhanlonii** [comment]
Thanks for the review @RSNara, I pushed all your updates 👍

**Pull request closed without merging**